src/main/java/uk/gov/hmcts/juror/api/moj/service/DeferralResponseServiceImpl.java

```diff
@@ -101,22 +101,25 @@ public void respondToDeferralRequest(BureauJwtPayload payload, DeferralRequestDt
     private void declineDeferralForJurorPool(BureauJwtPayload payload, DeferralRequestDto deferralRequestDto,
                                              JurorPool jurorPool) {
 
+        boolean isInDeferralMaintenance = jurorPool.getStatus().getStatus() == IJurorStatus.DEFERRED;
+        final Juror juror = jurorPool.getJuror();
         String username = payload.getLogin();
-
-        JurorStatus jurorStatus = new JurorStatus();
-        jurorStatus.setStatus(IJurorStatus.RESPONDED);
-        jurorPool.setStatus(jurorStatus);
-        jurorPool.setUserEdtq(username);
-        jurorPool.setDeferralCode(deferralRequestDto.getDeferralReason());
-        jurorPool.setDeferralDate(null);
-        jurorPool.setNextDate(jurorPool.getPool().getReturnDate());
-        jurorPoolRepository.save(jurorPool);
-
-        Juror juror = jurorPool.getJuror();
-        juror.setResponded(true);
-        juror.setUserEdtq(username);
+        //If juror is in deferral maintenance, we should not revert the existing (granted) deferral
+        if (!isInDeferralMaintenance) {
+            JurorStatus jurorStatus = new JurorStatus();
+            jurorStatus.setStatus(IJurorStatus.RESPONDED);
+            jurorPool.setStatus(jurorStatus);
+            jurorPool.setUserEdtq(username);
+            jurorPool.setDeferralCode(deferralRequestDto.getDeferralReason());
+            jurorPool.setDeferralDate(null);
+            jurorPool.setNextDate(jurorPool.getPool().getReturnDate());
+            jurorPoolRepository.save(jurorPool);
+
+            juror.setResponded(true);
+            juror.setUserEdtq(username);
```

Comment:
we probably want this line as well regardless as this shows which user edited the juror record

Reply:
Agreed updated